packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -192,13 +219,53 @@ impl Abigen {
         })
     }
 
+    pub fn includes(&self) -> Result<TokenStream, Error> {
+        let includes = match self.no_std {
+            true => quote! {
+                use alloc::{vec, vec::Vec};
+                use fuels_core::{EnumSelector, Parameterize, Tokenizable, Token, Identity, try_from_bytes};
+                use fuels_core::types::*;
+                use fuels_core::code_gen::function_selector::resolve_fn_selector;
+                use fuels_types::errors::Error as SDKError;
+                use fuels_types::param_types::{ParamType, EnumVariants};
+            },
+            false => quote! {
+                use fuels::contract::contract::{Contract, ContractCallHandler};
+                 use fuels::core::{EnumSelector, StringToken, Parameterize, Tokenizable, Token,
+                                  Identity, try_from_bytes};
+                use fuels::core::code_gen::{extract_and_parse_logs, extract_log_ids_and_data};
+                use fuels::core::abi_decoder::ABIDecoder;
+                use fuels::core::code_gen::function_selector::resolve_fn_selector;
+                use fuels::core::types::*;
+                use fuels::signers::WalletUnlocked;
+                use fuels::tx::{ContractId, Address, Receipt};
+                use fuels::types::bech32::Bech32ContractId;
+                use fuels::types::ResolvedLog;
+                use fuels::types::errors::Error as SDKError;
+                use fuels::types::param_types::{EnumVariants, ParamType};
+                use std::str::FromStr;
+                use std::collections::{HashSet, HashMap};
+            },
+        };
+        Ok(includes)
+    }
+
     pub fn functions(&self) -> Result<TokenStream, Error> {
-        let tokenized_functions = self
-            .abi
-            .functions
-            .iter()
-            .map(|function| expand_function(function, &self.types))
-            .collect::<Result<Vec<TokenStream>, Error>>()?;
+        let tokenized_functions = match self.is_script {
+            false => self
+                .abi
+                .functions
+                .iter()
+                .map(|function| expand_function(function, &self.types))
+                .collect::<Result<Vec<TokenStream>, Error>>()?,
+            true => self
+                .abi
+                .functions
+                .iter()
+                .filter(|function| function.name == "main")
```

Comment:
Curious, is a script valid if doesn't include a "main" method? Or in other words, what should be the expected behaviour on our side if not main is fond here?

Reply:
Trying to `forc build` a script that doesn't contain a `main` method yields: 
`Script declaration contains no main function. Scripts require a main function.`. 
I think we should probably `panic`, maybe?

Reply:
So I actually return an `InvalidData` error when the only method of a script is different from `main` (inside the `generate_script_argument_encoding_function`.